test/unit/org/apache/cassandra/cql3/ast/Select.java

```diff
@@ -479,17 +480,31 @@ public Builder table(TableMetadata table)
     public static class TableBasedBuilder extends BaseBuilder<TableBasedBuilder> implements Conditional.ConditionalBuilderPlus<TableBasedBuilder>
     {
         private final TableMetadata metadata;
+        private final ImmutableUniqueList<Symbol> columns;
 
         public TableBasedBuilder(TableMetadata metadata)
         {
             this.metadata = metadata;
             source = Optional.of(TableReference.from(metadata));
+            var builder = ImmutableUniqueList.<Symbol>builder();
+            metadata.allColumnsInSelectOrder().forEachRemaining(c -> builder.add(Symbol.from(c)));
+            columns = builder.buildAndClear();
         }
 
         @Override
         public TableMetadata metadata()
         {
             return metadata;
         }
+
+        private Symbol find(String name)
+        {
+            return columns.stream().filter(s -> s.symbol.equals(name)).findAny().get();
+        }
+
+        public TableBasedBuilder columnSelection(String name)
```

Comment:
This is unused. Just an FYI any data structures we additionally build here end up being created in production scenarios where prepared statements aren't used so there is an incentive to not build ones that aren't absolutely needed.